packages/headless/src/api/service/insight/insight-api-client.ts

```diff
@@ -0,0 +1,106 @@
+import {Logger} from 'pino';
+import {PlatformClient} from '../../platform-client';
+import {PreprocessRequest} from '../../preprocess-request';
+import {buildAPIResponseFromErrorOrThrow} from '../../search/search-api-error-response';
+import {
+  buildGetInsightInterfaceRequest,
+  GetInsightInterfaceRequest,
+} from './get-interface/get-interface-request';
+import {GetInsightInterfaceResponse} from './get-interface/get-interface-response';
+import {
+  buildInsightQueryRequest,
+  InsightQueryRequest,
+} from './query/query-request';
+import {InsightQueryResponse} from './query/query-response';
+import {
+  buildInsightUserActionsRequest,
+  InsightUserActionsRequest,
+} from './user-actions/user-actions-request';
+import {InsightUserActionsResponse} from './user-actions/user-actions-response';
+
+/**
+ * Initialization options for the `InsightAPIClient`.
+ */
+interface InsightAPIClientOptions {
+  logger: Logger;
+  preprocessRequest: PreprocessRequest;
+}
+
+export type InsightAPIResponse<TSuccessContent> =
+  | InsightAPISuccessResponse<TSuccessContent>
+  | InsightAPIErrorResponse;
+
+export interface InsightAPISuccessResponse<TContent> {
+  success: TContent;
+}
+
+export interface InsightAPIErrorStatusResponse {
+  statusCode: number;
+  message: string;
+  type: string;
+}
+
+export interface InsightAPIErrorResponse {
+  error: InsightAPIErrorStatusResponse;
+}
+
+/**
+ * The client to use to interact with the Insight API.
+ */
+export class InsightAPIClient {
+  constructor(private options: InsightAPIClientOptions) {}
+
+  async getInterface(
+    req: GetInsightInterfaceRequest
+  ): Promise<InsightAPIResponse<GetInsightInterfaceResponse>> {
+    const response = await PlatformClient.call({
+      ...buildGetInsightInterfaceRequest(req),
+      ...this.options,
+    });
+
+    if (response instanceof Error) {
+      return buildAPIResponseFromErrorOrThrow(response);
```

Comment:
Since it's a different API (and not a search API error), I'm not certain this will always be the appropriate return type. 

Unless the Insight panel API always copies the Search API error format.

Reply:
So far the Customer Service API is returning the error body in the same format as the Search API. We're doing the same thing already for the Case Assist API client.